test/test_torch.py

```diff
@@ -6843,7 +6843,7 @@ def inner(self, device, dtype):
         1e-5, 1e-5, 1e-5, torch.testing.get_all_dtypes(include_complex=False, include_bool=False), _cpu_types, False),
     ('topk', 'dim_desc_sort', _small_3d_unique, lambda t, d: [2, 1, True, True],
         1e-5, 1e-5, 1e-5, torch.testing.get_all_dtypes(include_complex=False, include_bool=False), _cpu_types, False),
-    ('trace', '', _medium_2d, lambda t, d: [], 1e-3, 1e-5, 1e-5, _types, _cpu_types, False),
+    ('trace', '', _medium_2d, lambda t, d: [], 1e-3, 1e-5, 1e-5, _types + _complex_types, _cpu_types, False),
```

Comment:
Implement an OpInfo for trace and delete this entry, update `test_trace` in test_shape_ops to include complex types, and delete its method_test entry:

https://github.com/pytorch/pytorch/blob/057be231688f712212688530f0105ba143e9a1bf/test/test_shape_ops.py#L228

https://github.com/pytorch/pytorch/blob/057be231688f712212688530f0105ba143e9a1bf/torch/testing/_internal/common_methods_invocations.py#L2221

Reply:
done!